applications/GeoMechanicsApplication/tests/cpp_tests/custom_elements/test_integration_schemes.cpp

```diff
@@ -0,0 +1,130 @@
+    };
+    KRATOS_EXPECT_TRUE(std::all_of(rIntegrationPoints.begin(), rIntegrationPoints.end(), xi_is_in_range))
+
+    auto non_xi_coordinates_must_be_near_zero = [tolerance](const auto& rPoint) {
+        return (std::abs(rPoint[1]) <= tolerance) && (std::abs(rPoint[2]) <= tolerance);
+    };
+    KRATOS_EXPECT_TRUE(std::all_of(rIntegrationPoints.begin(), rIntegrationPoints.end(),
+                                   non_xi_coordinates_must_be_near_zero))
+}
+
+} // namespace
+
+namespace Kratos::Testing
+{
+
+KRATOS_TEST_CASE_IN_SUITE(ALobattoIntegrationSchemeIsAnIntegrationScheme, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    constexpr auto number_of_points = 2;
+    const auto     scheme           = LobattoIntegrationScheme{number_of_points};
+
+    KRATOS_EXPECT_NE(dynamic_cast<const IntegrationScheme*>(&scheme), nullptr);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(NumberOfIntegrationPointsMatchesTheNumberOfPointsGivenAtConstructionTime,
+                          KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    const auto supported_numbers_of_points = std::vector<std::size_t>{2, 3};
+
+    for (auto number : supported_numbers_of_points) {
+        const auto scheme = MakeLobattoIntegrationScheme(number);
+
+        KRATOS_EXPECT_EQ(scheme->GetNumberOfIntegrationPoints(), number);
+        KRATOS_EXPECT_EQ(scheme->GetIntegrationPoints().size(), number);
+    }
+}
+
+KRATOS_TEST_CASE_IN_SUITE(SumOfIntegrationPointWeightsOfAllSupportedLobattoSchemesEqualsTwo,
+                          KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    const auto supported_numbers_of_points = std::vector<std::size_t>{2, 3};
+
+    constexpr auto relative_tolerance = 1.0e-6;
+    for (auto number : supported_numbers_of_points) {
+        const auto scheme = MakeLobattoIntegrationScheme(number);
+
+        KRATOS_EXPECT_RELATIVE_NEAR(SumOfWeights(scheme->GetIntegrationPoints()), 2.0, relative_tolerance)
+    }
+}
+
+KRATOS_TEST_CASE_IN_SUITE(PointsOfAllSupportedLobattoSchemesMustBeInRangeAndIncludeBounds,
+                          KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    const auto supported_numbers_of_points = std::vector<std::size_t>{2, 3};
+
+    for (auto number : supported_numbers_of_points) {
+        const auto scheme = MakeLobattoIntegrationScheme(number);
+
+        ExpectLocalCoordinatesAreInRange(scheme->GetIntegrationPoints());
+        ExpectLocalCoordinatesIncludeRangeBounds(scheme->GetIntegrationPoints());
+    }
+}
+
+KRATOS_TEST_CASE_IN_SUITE(AttemptingToMakeALobattoSchemeWithAnUnsupportedNumberOfPointsRaisesAnError,
+                          KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    const auto some_unsupported_numbers_of_points = std::vector<std::size_t>{0, 1, 4, 5, 6, 7};
+
+    for (auto number : some_unsupported_numbers_of_points) {
+        const auto expected_error_message =
+            "Can't construct Lobatto integration scheme: no support for " + std::to_string(number) + " point(s)";
```

Comment:
The header in which std::to_string is defined is not included here. That seems to hold for more of the things like std::vector, std::make_unique, std::any_of etc.

Reply:
Good catch. To make the test code more robust, I have included some STL headers.